transports/webrtc-websys/src/stream.rs

```diff
@@ -0,0 +1,268 @@
+                    if let Some(message) = message {
+                        *read_buffer = message.into();
+                    }
+                    // continue to loop
+                }
+                None => {
+                    state.handle_inbound_flag(Flag::FIN, read_buffer);
+                    return Poll::Ready(Ok(0));
+                }
+            }
+        }
+    }
+}
+
+impl AsyncWrite for Stream {
+    /// Attempt to write bytes from buf into the object.
+    /// On success, returns Poll::Ready(Ok(num_bytes_written)).
+    /// If the object is not ready for writing,
+    /// the method returns Poll::Pending and
+    /// arranges for the current task (via cx.waker().wake_by_ref())
+    /// to receive a notification when the object becomes writable
+    /// or is closed.
+    ///
+    /// In WebRTC DataChannels, we can always write to the channel
+    /// so we don't need to poll the channel for writability
+    /// as long as the channel is open
+    ///
+    /// So we need access to the channel or peer_connection,
+    /// and the State of the Channel (DataChannel.readyState)
+    fn poll_write(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context,
+        buf: &[u8],
+    ) -> Poll<io::Result<usize>> {
+        while self.state.read_flags_in_async_write() {
+            // TODO: In case AsyncRead::poll_read encountered an error or returned None earlier, we will poll the
+            // underlying I/O resource once more. Is that allowed? How about introducing a state IoReadClosed?
+
+            let Self {
+                read_buffer,
+                state,
+                inner,
+                ..
+            } = &mut *self;
+
+            match io_poll_next(&mut inner.io, cx)? {
+                Poll::Ready(Some((Some(flag), message))) => {
+                    // Read side is closed. Discard any incoming messages.
+                    drop(message);
+                    // But still handle flags, e.g. a `Flag::StopSending`.
+                    state.handle_inbound_flag(flag, read_buffer)
+                }
+                Poll::Ready(Some((None, message))) => drop(message),
+                Poll::Ready(None) | Poll::Pending => break,
+            }
+        }
+
+        self.state.write_barrier()?;
+
+        ready!(self.inner.io.poll_ready_unpin(cx))?;
+
+        let n = usize::min(buf.len(), MAX_DATA_LEN);
+
+        Pin::new(&mut self.inner.io).start_send(Message {
+            flag: None,
+            message: Some(buf[0..n].into()),
+        })?;
+
+        Poll::Ready(Ok(n))
+    }
+
+    /// Attempt to flush the object, ensuring that any buffered data reach their destination.
+    ///
+    /// On success, returns Poll::Ready(Ok(())).
+    ///
+    /// If flushing cannot immediately complete, this method returns Poll::Pending and
+    /// arranges for the current task (via cx.waker().wake_by_ref()) to receive a
+    /// notification when the object can make progress towards flushing.
```

Comment:
```suggestion
```

I don't think there is much value in these general trait comments.

Reply:
Removed.